providers/ipinfo/resources/ipinfo.go

```diff
@@ -0,0 +1,296 @@
+func createHTTPClientForInterface(interfaceIP net.IP) (*http.Client, error) {
+	// Find the interface that has this IP
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to list interfaces")
+	}
+
+	var targetIP net.IP
+	var ifaceName string
+	for _, iface := range ifaces {
+		addrs, err := iface.Addrs()
+		if err != nil {
+			continue
+		}
+		for _, addr := range addrs {
+			var ip net.IP
+			switch v := addr.(type) {
+			case *net.IPNet:
+				ip = v.IP
+			case *net.IPAddr:
+				ip = v.IP
+			}
+			if ip != nil && ip.Equal(interfaceIP) {
+				// Found the interface, use this IP for binding
+				targetIP = ip
+				ifaceName = iface.Name
+				break
+			}
+		}
+		if targetIP != nil {
+			break
+		}
+	}
+
+	if targetIP == nil {
+		return nil, errors.New("interface with IP not found")
+	}
+
+	// Create TCP address with the IP (port 0 means let OS choose)
+	// For IPv6 link-local addresses, we need to use the zone identifier
+	var localAddr *net.TCPAddr
+	if targetIP.To4() != nil {
+		localAddr = &net.TCPAddr{IP: targetIP.To4(), Port: 0}
+	} else {
+		// For IPv6, check if it's link-local and needs zone identifier
+		if targetIP.IsLinkLocalUnicast() && ifaceName != "" {
+			// Try to resolve with zone identifier
+			// Format: fe80::1%en0
+			zoneAddr := targetIP.String() + "%" + ifaceName
+			ipAddr, err := net.ResolveIPAddr("ip6", zoneAddr)
+			if err == nil {
+				localAddr = &net.TCPAddr{IP: ipAddr.IP, Zone: ifaceName, Port: 0}
+			} else {
+				// Fallback to IP without zone (may not work for link-local)
+				localAddr = &net.TCPAddr{IP: targetIP.To16(), Port: 0}
+			}
+		} else {
+			localAddr = &net.TCPAddr{IP: targetIP.To16(), Port: 0}
+		}
+	}
+
+	// Create a custom dialer that binds to the specific interface
+	dialer := &net.Dialer{
+		LocalAddr: localAddr,
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}
+
+	// Create a transport with the custom dialer
+	transport := &http.Transport{
+		DialContext:       dialer.DialContext,
+		DisableKeepAlives: false,
+	}
+
+	// Create an HTTP client with the custom transport
+	client := &http.Client{
+		Transport: transport,
+		Timeout:   30 * time.Second,
+	}
+
+	return client, nil
+}
+
+// ipinfoResponse represents the JSON response from ipinfo.io API
+type ipinfoResponse struct {
+	IP       string `json:"ip"`
+	Hostname string `json:"hostname"`
+	City     string `json:"city"`
+	Region   string `json:"region"`
+	Country  string `json:"country"`
+	Loc      string `json:"loc"`
+	Org      string `json:"org"`
+	Postal   string `json:"postal"`
+	Timezone string `json:"timezone"`
+}
+
+// callIpinfoAPI makes a direct HTTP call to ipinfo.io API
+func callIpinfoAPI(client *http.Client, queryIP net.IP, token string) (*ipinfoResponse, error) {
+	var url string
+	if queryIP == nil {
+		// Query for YOUR public IP
+		url = "https://ipinfo.io"
+	} else {
+		// Query for specific IP
+		url = fmt.Sprintf("https://ipinfo.io/%s", queryIP.String())
```

Comment:
any reason why we are not using https://github.com/ipinfo/go? This would also help us to implement the token support.

Reply:
Empty token works without any problem:

```go
const token = ""

// params: httpClient, cache, token. `http.DefaultClient` and no cache will be used in case of `nil`.
client := ipinfo.NewClient(nil, nil, token)

const ip_address = "8.8.8.8"
info, err := client.GetIPInfo(net.ParseIP(ip_address))

if err != nil {
panic(err)
}

fmt.Println(info)
```